test/onigumo_test.exs

```diff
@@ -20,19 +24,71 @@ defmodule OnigumoTest do
       end
     )
 
-    assert(:ok == Onigumo.download(HTTPoisonMock, @url))
-    assert("Body from: #{@url}" == File.read!(@filename))
+    url = Enum.at(@urls, 0)
+    assert(:ok == Onigumo.download(HTTPoisonMock, url))
+    assert("Body from: #{url}" == File.read!(@filename))
   end
 
 
+  test("download multiple URLs") do
+    expect(
+      HTTPoisonMock,
+      :get!,
+      2,
+      fn url ->
+        %HTTPoison.Response{
+          status_code: 200,
+          body: "Body from: #{url}"
+        }
+      end
+    )
+
+    assert([:ok, :ok] == Onigumo.download(HTTPoisonMock, @urls))
+    last_url = Enum.at(@urls, -1)
+    assert("Body from: #{last_url}" == File.read!(@filename))
+  end
+
+  test("download URLs from the input file") do
+    expect(
+      HTTPoisonMock,
+      :get!,
+      length(@urls),
+      fn url ->
+        %HTTPoison.Response{
+          status_code: 200,
+          body: "Body from: #{url}"
+        }
+      end
+    )
+
+    content = Enum.map(@urls, &(&1 <> "\n")) |> Enum.join()
+    File.write!(@testfile_with_urls, content)
+
+    last_url = Enum.at(@urls, -1)
+    expected = "Body from: #{last_url}"
+    Onigumo.download(HTTPoisonMock)
+
+    assert(expected == File.read!(@filename))
+  end
+
   @tag :tmp_dir
-  test("load URL from file", %{tmp_dir: tmp_dir}) do
+  test("load a single URL from a file", %{tmp_dir: tmp_dir}) do
+    url = Enum.at(@urls, 0)
+
     filepath = Path.join(tmp_dir, @testfile_with_urls)
-    content = @url <> " \n"
+    content = url <> " \n"
     File.write!(filepath, content)
 
-    expected = [@url]
+    expected = [url]
     assert(expected == Onigumo.load_urls(filepath))
   end
 
+  @tag :tmp_dir
+  test("load multiple URLs from a file", %{tmp_dir: tmp_dir}) do
+    filepath = Path.join(tmp_dir, @testfile_with_urls)
+    content = Enum.map(@urls, &(&1 <> "\n")) |> Enum.join()
+    File.write!(filepath, content)
+
+    assert(@urls == Onigumo.load_urls(filepath))
+  end
```

Comment:
This test is the only new test that does not require the code changes. It could have been in the original code or in a separate pull request.

Reply:
Extracted to #47.